Configuration/Applications/python/ConfigBuilder.py

```diff
@@ -2160,6 +2153,11 @@ def prepare(self, doChecking = False):
 
 	self.pythonCfgCode += result
 
+        from PhysicsTools.PatAlgos.tools.helpers import schedulePatAlgosEndPath
+        schedulePatAlgosEndPath(self.process)
+        self.pythonCfgCode+="from PhysicsTools.PatAlgos.tools.helpers import schedulePatAlgosEndPath\n"
+        self.pythonCfgCode+="schedulePatAlgosEndPath(process)\n"
+
```

Comment:
No, it did not have this end path before. I had to associate the task with some path, end path or the top level schedule that was guaranteed to exist. I did not find such a thing so I created one. That function in helpers.py will create one task and one end path if it has not already been created and then attach it to the schedule. I am open to suggestions as to a better way to do this, but this does work. The "tools" in PAT often create EDProducers and attach them to the process and there  is often no obvious task to attach the EDProducer to or sequence to attach the task to. Usually what is available in the tool function is the process.